vars/buildNodeJs.groovy

```diff
@@ -52,17 +57,30 @@ def call(Map config=[:]) {
         config['buildClosure']()
       }
 
-      stage('Run tests') {
+      if (fileExists('./docker-compose.test.yaml')) {
+        stage('Run tests') {
         build.runTests(repoName, repoName, BUILD_NUMBER, tag, pr, config.environment)
       }
-
-      stage('Create JUnit report') {
+      } else {
+      echo("docker-compose.test.yaml not found, skipping test run")
+      }
+      
+      if (fileExists('./docker-compose.test.yaml')) {
```

Comment:
You can put both the next steps in the same if statement to save duplication.

Also remove else.